src/lib/pickles/fix_domains.ml

```diff
@@ -7,14 +7,54 @@ let rough_domains =
   let d = Domain.Pow_2_roots_of_unity 20 in
   { Domains.h = d }
 
-let domains (type field) ?(min_log2 = 0)
+let domains (type field) ?feature_flags
     (module Impl : Snarky_backendless.Snark_intf.Run with type field = field)
```

Comment:
Please assert equality between `Constraint_system.t` and the `Plonk_constraint_system.t` instance instead of changing the snarky interface.

Reply:
That sounds a good idea! Let me try.

Reply:
First solution in 719c23c88deff74edd02d9fae1fa82d8d831a431. Let me know if it is ok for you. I try to add it outside of the signature, and in `Pallas_constraint_system`/`Vesta_constraint_system`.

Reply:
Seems to be a clean solution. Closing the thread.